File: backend/src/main/scala/microservices/content/objects/CreateReviewPlannerRequest.scala
package com.typesafe.travel.content.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class CreateReviewPlannerRequest(
    userId: String,
    orderId: String,
    orderItemId: String,
    rating: Int,
    title: String,
    content: String,
    images: List[ContentImagePlannerResponse]
)
object CreateReviewPlannerRequest:
  given sourceEncoder: Encoder[CreateReviewPlannerRequest] = deriveEncoder
  given sourceDecoder: Decoder[CreateReviewPlannerRequest] = deriveDecoder

File: frontend/src/microservices/content/objects/CreateReviewPlannerRequest.ts
// CreateReviewPlannerRequest：content 域创建评论请求对象。

import type { ContentImagePlannerResponse } from './ContentImagePlannerResponse'

export type CreateReviewPlannerRequest = {
  userId: string
  orderId: string
  orderItemId: string
  rating: number
  title: string
  content: string
  images: ContentImagePlannerResponse[]
}

export const createReviewPlannerRequestFromJson = (json: string): CreateReviewPlannerRequest =>
  JSON.parse(json) as CreateReviewPlannerRequest

export const createReviewPlannerRequestToJson = (value: CreateReviewPlannerRequest): string =>
  JSON.stringify(value)
